core/include/cubos/core/net/udp_socket.hpp

```diff
@@ -0,0 +1,81 @@
+/// @file
+/// @brief Class @ref cubos::core::net::UdpSocket.
+/// @ingroup core-net
+
+#pragma once
+
+#include <cstdint>
+#include <optional>
+#include <string>
+
+#include <cubos/core/api.hpp>
+#include <cubos/core/net/address.hpp>
+
+#ifdef _WIN32
+#include <vcruntime.h>
+#include <winsock2.h>
```

Comment:
Are you sure you need to include this here? I'm afraid of macro troubles down the line.
Is the type of `SOCKET` always the same? If so, you can just use that type down below.
Also, if you change this, avoid the `#define`s. Having them on the .cpp is fine though.